synapse/storage/_base.py

```diff
@@ -57,7 +57,21 @@ def process_replication_rows(  # noqa: B027 (no-op by design)
         token: int,
         rows: Iterable[Any],
     ) -> None:
-        pass
+        """
+        Used by storage classes to invalidate caches based on incoming replication data.
+        """
```

Comment:
Can you mention that we should *not* be advancing stream ID generators here please? Just to spell out which function should be used for what.

Reply:
https://github.com/matrix-org/synapse/pull/14723/commits/537cb404678f050db9d00262bdeb86bee6dfba67